### ISSUE TITLE
NPE in AddressController (0.17.1-rc1)

### ISSUE THREAD

**Issue by calohmn** [EnMasseProject/enmasse#1018]
With 0.17.1-rc1, we got this exception in the AddressController:
````
2018-03-07 14:55:15.094 [Thread-4] DEBUG ControllerChain:99 - Check standard address spaces: [{name=default,namespace=hono,type=standard,plan=unlimited-standard,endpoints=[]}]
2018-03-07 14:55:15.099 [Thread-4] WARN  ResourceController:77 - Exception doing resource update
java.lang.NullPointerException
	at io.enmasse.address.model.AddressSpaceResolver.getType(AddressSpaceResolver.java:20)
	at io.enmasse.address.model.AddressSpaceResolver.validate(AddressSpaceResolver.java:33)
	at io.enmasse.controller.CreateController.handle(CreateController.java:43)
	at io.enmasse.controller.ControllerChain.resourcesUpdated(ControllerChain.java:104)
	at io.enmasse.k8s.api.ResourceController.run(ResourceController.java:72)
	at java.lang.Thread.run(Thread.java:748)
````
The 3rd log output of "Check standard address spaces" isn't followed by the exception anymore.

I guess here the `CachingSchemaProvider` hasn't gotten the schema yet via `resourcesUpdated`.

**Comment by lulf** [comment]
I think that’s an accurate observation. The code should handle it a bit more gracefully

**Comment by calohmn** [comment]
Here a corresponding exception in the standard-controller container:
````
2018-03-07 16:16:52.686 [Thread-4] INFO  AddressController:82 - resourcesUpdated: Check addresses in address space controller: []
2018-03-07 16:16:52.687 [Thread-4] WARN  ResourceController:77 - Exception doing resource update
java.lang.NullPointerException
	at io.enmasse.controller.standard.AddressController.resourcesUpdated(AddressController.java:85)
	at io.enmasse.k8s.api.ResourceController.run(ResourceController.java:72)
	at java.lang.Thread.run(Thread.java:748)
````

**Comment by lulf** [comment]
Fixed in 0.18.0
